frontend/src/components/forms/SelectField.tsx

```diff
@@ -0,0 +1,99 @@
+import {
+  FunctionComponent,
+  useState,
+  ReactNode,
+  ChangeEvent,
+  InputHTMLAttributes,
+} from 'react'
+import { nanoid } from 'nanoid'
+import SelectInput from './SelectInput'
+import Label from './Label'
+import InlineError from './InlineError'
+import { FormRegisterType } from '../../types/form-types'
+
+export interface SelectOption {
+  label: ReactNode
+  value: string | number
+  disabled?: boolean
+}
+
+type Props = InputHTMLAttributes<HTMLSelectElement> & {
+  /**
+   * The ID of the input, used to map the label to the input element
+   */
+  id?: string
+  /**
+   * A required label for the field
+   */
+  label: ReactNode
+  /**
+   * A name added to the input element can be used for form parsing and
+   * validation
+   */
+  name?: string
+  /**
+   * If true, the input will not be editable, but the text will remain
+   * selectable
+   */
+  readOnly?: boolean
+  /**
+   * A string that represents a custom validation error
+   */
+  error?: string
+  /**
+   * The type of the input element. Defaults to "text"
+   */
+  type?: string
+  /**
+   * If true, the input field will not be editable or selectable
+   */
+  disabled?: boolean
+  /**
+   * A required list of options to display in the dropdown
+   */
+  options: SelectOption[]
+  /**
+   * Callback triggered when the text changes. Works the same way as a regular
+   * input element.
+   */
+  onChange?: (event: ChangeEvent<HTMLSelectElement>) => void
+  /**
+   * The register function from `react-hook-form`'s `useForm()` hook used for validation and submission
+   */
+  register?: FormRegisterType
+}
+
+const SelectField: FunctionComponent<Props> = ({
+  id,
+  label,
+  error = null,
+  options = [],
+  ...otherProps
+}) => {
+  // Create a unique ID in case the use doesn't provide one
+  const [uniqueId] = useState(nanoid())
```

Comment:
Thought (non-blocking): it's still weird to me that hooks force this thing where you must call them regardless of whether or not you'll use them

Reply:
🤔 we are using the hook here though. It's storing the `uniqueId` variable so that it doesn't change between every render.

Reply:
Not if `id` is present though, yeah?

Reply:
Right, if an `id` prop is explicitly passed, the `fieldId` variable below will use that instead. But either way, the `uniqueId` variable is instantiated once the component is mounted, and never changes after that.